### PR TITLE
[SDKS-6389] Add localhost JSON sanitization

### PR DESCRIPTION
**Pull request by nmayorsplit**

### CODE REVIEW

File: client/pom.xml [hunk 1]
```diff
@@ -177,6 +177,11 @@
             <artifactId>snakeyaml</artifactId>
             <version>1.32</version>
         </dependency>
+        <dependency>
```
Comment (mmelograno) [inline client/pom.xml:180]: why we need a new dependency for parsing JSON?

File: client/src/main/java/io/split/client/LocalhostSegmentChangeFetcher.java [hunk 2]
```diff
@@ -24,8 +24,9 @@ public LocalhostSegmentChangeFetcher(String filePath){
     @Override
     public SegmentChange fetch(String segmentName, long changesSinceThisChangeNumber, FetchOptions options) {
         try {
-            JsonReader jsonReader = new JsonReader(new FileReader(String.format("%s/%s.json", _file, segmentName)));
-            return Json.fromJson(jsonReader, SegmentChange.class);
+            ObjectMapper objectMapper = new ObjectMapper();
```
Comment (mmelograno) [inline client/src/main/java/io/split/client/LocalhostSegmentChangeFetcher.java:27]: JsonReader jsonReader = new JsonReader(new FileReader(String.format("%s/%s.json", _file, segmentName)));
            SegmentChange segmentChange = Json.fromJson(jsonReader, SegmentChange.class);
            return LocalhostSanitizer.sanitization(segmentChange);

File: client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java [hunk 3]
```diff
@@ -23,9 +23,11 @@ public LocalhostSplitChangeFetcher(String filePath){
 
     @Override
     public SplitChange fetch(long since, FetchOptions options) {
+
         try {
-            JsonReader jsonReader = new JsonReader(new FileReader(_file));
-            return Json.fromJson(jsonReader, SplitChange.class);
+            ObjectMapper objectMapper = new ObjectMapper();
```
Comment (mmelograno) [inline client/src/main/java/io/split/client/LocalhostSplitChangeFetcher.java:28]: JsonReader jsonReader = new JsonReader(new FileReader(_file));
            SplitChange splitChange = Json.fromJson(jsonReader, SplitChange.class);
            return LocalhostSanitizer.sanitization(splitChange);

**Review by mmelograno — CHANGES_REQUESTED** [review]

File: client/src/main/java/io/split/client/utils/LocalhostSanitizer.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                    split.conditions = new ArrayList<>();
+                }
+
+                Condition condition = new Condition();
+                if (!split.conditions.isEmpty()){
+                     condition = split.conditions.get(split.conditions.size() - 1);
+                }
+
+                if (split.conditions.isEmpty() || !condition.conditionType.equals(ConditionType.ROLLOUT) ||
+                        !condition.matcherGroup.matchers.get(0).matcherType.equals(MatcherType.ALL_KEYS)) {
```
Comment (mmelograno) [inline client/src/main/java/io/split/client/utils/LocalhostSanitizer.java:80]: missing matchers empty before accessing

**Review by mmelograno — CHANGES_REQUESTED** [review]

File: client/pom.xml (hunk 1 above)
Comment (nmayorsplit) [reply client/pom.xml:180]: We need that one to map the empty elements or elements that aren't.

File: client/src/main/java/io/split/client/LocalhostSegmentChangeFetcher.java (hunk 2 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/LocalhostSegmentChangeFetcher.java:27]: In this way, it's not going to work. If an element isn't in the JSON file is going to throw an exception.

File: client/src/main/java/io/split/client/utils/LocalhostSanitizer.java (hunk 4 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/utils/LocalhostSanitizer.java:82]: I added the new check.

**Review by mmelograno — APPROVED** [review]

**Pull request merged**